parser/prism/Helpers.h

```diff
@@ -0,0 +1,22 @@
+#ifndef SORBET_PARSER_PRISM_HELPERS_H
+#define SORBET_PARSER_PRISM_HELPERS_H
+
+extern "C" {
+#include "prism.h"
+}
+
+namespace sorbet::parser::Prism {
+
+// Take a pointer to a Prism node "subclass" (a thing with an embedded `pm_node_t base` as its first member),
+// and up-casts it back to a general `pm_node_t` pointer.
+template <typename PrismNode> pm_node_t *up_cast(PrismNode *node) {
+    static_assert(!std::is_same_v<PrismNode, pm_node_t>,
+                  "There's no need to call `up_cast` here, because this is already a `pm_node_t`.");
+    static_assert(std::is_same_v<decltype(PrismNode::base), pm_node_t>,
```

Comment:
This is saying that whatever type `PrismNode`, it has to have a `base` field whose type is `pm_node_t`.

This is a decent approximation of 'Is this `PrismNode` a Prism node "subclass"?', since all such node types start with an embedded `pm_node_t base` field.

Reply:
I extracted this into a `isPrismNode` for better clarity.